File: tests/split_sec.js
let mod = require('../number.js').mod;

function horners(x, k, field, poly, secret) {
        let result = poly[0];
        // evaluate polynomial with horner's method
        for(let i = 1; i < k; i++)
                        result = mod(mod(result * x, field) + poly[i], field);
        return result;
}

function split_secret(n, k, field, coeffs, secret) {
        let shares = [];
        let s = 0;
        for (let i = 0; i < n; i++) {
                s = horners(i+1, k, field, coeffs, secret);
                shares.push(s);
        }
        return shares;
}

let n = 2;
let k = 2;
let field = 257;
let coeffs = [222,34,5,61,19];
let secret = 101
console.log(split_secret(n,k,field,coeffs,secret));

File: tests/split_sec.py
def _horners(x, k, field, coeff, secret):
    ''' Evaluate the polynomial with Horner's method '''
    res = coeff[0]
    for i in range(1, k):
        res = ((res * x) % field + coeff[i]) % field
    return res

def _split_secret(n, k, field, coeffs, secret):
    ''' Split secret into shares '''
    shares = []
    s = 0
    for i in range(n):
        s = _horners(i+1, k, field, coeffs, secret)
        shares.append(s)
    return shares


n = 2
k = 2
field = 257
coeff = [222,34,5,61,19]
secret = 101
print(_split_secret(n,k,field,coeff,secret))
